File: src/activity/Time.java
package activity;

class Time {

    static String time(double seconds) {
        int s = (int) seconds;
        int m = 0;
        int h = 0;

        while (s >= 60) {
            s -= 60;

            m++;
        }

        while (m >= 60) {
            m -= 60;

            h++;
        }



        var str = "";

        if (h > 0) {
            if (h == 1)
                str = "1 hour";
            else
                str = h + " hours";


            if (m == 1)
                str += " 1 minute";
            else if (m > 1)
                str += " " + m + " minutes";
        } else if (m > 0) {
            if (m == 1)
                str += " 1 minute";
            else
                str += " " + m + " minutes";


            if (s == 1)
                str += " 1 second";
            else if (s > 1)
                str += " " + s + " seconds";
        } else if (s == 1)
            str = "1 second";
        else
            str = s + " seconds";

        return str;
    }

    static String time(double seconds, boolean everything) {
        int s = (int) seconds;
        int m = 0;
        int h = 0;

        while (s >= 60) {
            s -= 60;

            m++;
        }

        while (m >= 60) {
            m -= 60;

            h++;
        }


        if (everything) {
            var str = "";

            if (h > 0) {
                if (h == 1)
                    str = "1 hour ";
                else
                    str = h + " hours ";
            }
            if (m > 0) {
                if (m == 1)
                    str += "1 minute ";
                else
                    str += " " + m + " minutes ";
            }
            if (s > 0) {
                if (s == 1)
                    str += "1 second";
                else
                    str += s + " seconds";
            }

            return str;
        } else
            return time(seconds);
    }
}

File: src/activity/Time.scala
package activity

object Time {

    private def getTime(seconds: Double): (Int, Int, Int) = {
        var s = seconds.toInt
        var m = 0
        var h = 0

        while (s >= 60) {
            s -= 60
            m += 1
        }
        while (m >= 60) {
            m -= 60
            h += 1
        }

        (h, m, s)
    }

    def time(seconds: Double): String = {
        val (h, m, s) = getTime(seconds)

        var str = ""

        if (h > 0) {
            str = s"$h hour${if (h != 1) "s"}"
            if (m > 0) str += s" $m minute${if (m != 1) "s"}"
        }
        else if (m > 0) {
            str = s"$m minute${if (m != 1) "s"}"
            if (s > 0) str += s" $s second${if (s != 1) "s"}"
        }
        else str = s"$s second${if (s != 1) "s"}"

        str
    }

    def time(seconds: Double, everything: Boolean): String = {
        val (h, m, s) = getTime(seconds)

        if (everything) {
            var str = ""
            if (h > 0) str = s"$h hour${if (h != 1)"s"} "
            if (m > 0) str += s"$m minute${if (m != 1)"s"} "
            if (s > 0) str += s"$s second${if (s != 1)"s"} "
            str
        }
        else time(seconds)
    }
}
